ui/app/sample-data/benchmarkDashboard.ts

```diff
@@ -22,7 +22,7 @@ const benchmarkDashboard: DashboardResource = {
     updated_at: '2021-11-09',
   },
   spec: {
-    datasource: { kind: 'Prometheus', name: 'PrometheusDemo', global: true },
+    datasource: { $ref: '/api/v1/globaldatasources/PrometheusDemoBrowser#' },
```

Comment:
is it the regular syntax to have a `#` at the end of the URI ?

Reply:
Something just came to my mind:  how do we use the default datasource here ?

Currently you can have one default datasource per type of datasource. So I'm wondering how you can say I want to use the Prometheus default datasource.

Reply:
> is it the regular syntax to have a `#` at the end of the URI ?

Yeah, in json ref terms it just means "the root node of the document" because in theory this could actually be `/my/url#/path/in/document/to/datasource`. But maybe it's not necessary for the root in this case and it will default to it without the `#`? I can try it without and see if it works. 🤷‍♂️ 

> Something just came to my mind: how do we use the default datasource here ?
> 
> Currently you can have one default datasource per type of datasource. So I'm wondering how you can say I want to use the Prometheus default datasource.

Interesting. I was under the impression that there were two "defaults" in play:

1. The default for the dashboard, specified in the dashboard spec as `datasource`. Right now that field is required although I was thinking maybe it should be optional in which case it would fall back to...
2. The system-wide default datasource which the `DataSourceRegistry` is calculating by looking at the list of datasources and finding the one with the default flag (regardless of kind).

You think there should be a default per kind?

Reply:
> Yeah, in json ref terms it just means "the root node of the document" because in theory this could actually be /my/url#/path/in/document/to/datasource

Well if it's coming from the `jsonref` specification, we should follow it even it doesn't really make sense for us on this particular point.

> You think there should be a default per kind?

That was something I found cool. But it can be changed if it's too hard to managed on frontend side or totally overkilled.

In my mind saying, I want to use the default Prometheus datasource, and then default Log Server datasource for my dashboard sounds cool to me.

Reply:
So, if we wanted to do a "default per kind", I think we're back to the question we discussed in Matrix chat which is "how do you point at a Datasource"? I don't think using a json reference works in that case.

So how would I say, "I want to use a Prometheus Datasource" or "I want to use a Log Server Datasource". The previous thing we had with a "Datasource selector" was OK, but the syntax felt a little weird and the API when you query returned multiple Datasources, so it felt like there wasn't really a way to point at a specific one.

Reply:
BTW, I'm not at all tied to using json refs, that was just my suggestion for solving the problem of needing to point at a specific datasource. If some sort of "selector" API is better or more desirable, we should definitely do that and figure out how to make it possible to make sure you only get a single Datasource back when you use a selector.

Reply:
>  The previous thing we had with a "Datasource selector" was OK, but the syntax felt a little weird and the API when you query using the selector could potentially return multiple Datasources that matched the selector, so it felt like there wasn't really a way to point at a specific one

mmm the previous selector is this one:

```go
type Datasource struct {
	// Name is the name of the datasource
	Name string `json:"name" yaml:"name"`
	// Kind is the datasource kind
	Kind datasource.Kind `json:"kind" yaml:"kind"`
	// If global is true, we are referencing a global datasource.
	// When set to false, we are referencing a datasource in the same project as the current dashboard.
	Global bool `json:"global" yaml:"global"`
}
```

`Kind` is only necessary in case you would like to use the default datasource that is matching this type.

In case you want to specify the datasource you would like to use, you can give the name as it is the unique key.

Then if the value `Global` is set to false, then the datasource should be retrieved in the same project where the dashboard belongs.

But before going back to a selector or moving forward with the json ref, we should assess if:
* We want a default datasource
* We want a default datasource per kind.

Both are ok from my perspective. Grafana is doing the first one only if I remember correctly. 

I'm really sorry because I totally forgot about the `default` feature. But now it is the good time to assess if we want to keep it and to remove it :).

Last thing: if the selector wasn't clear at all how to use it, definitely we should change it to make it clearer.